spec/ameba/rule/style/redundant_parentheses_spec.cr

```diff
@@ -0,0 +1,115 @@
+require "../../../spec_helper"
+
+module Ameba::Rule::Style
+  subject = RedundantParentheses.new
+
+  describe RedundantParentheses do
+    {% for keyword in %w(if unless while until) %}
+      context "{{ keyword.id }}" do
+        it "reports if redundant parentheses are found" do
+          source = expect_issue subject, <<-CRYSTAL, keyword: {{ keyword }}
+            %{keyword}   (foo > 10)
+            _{keyword} # ^^^^^^^^^^ error: Redundant parentheses
+              foo
+            end
+            CRYSTAL
+
+          expect_correction source, <<-CRYSTAL
+            {{ keyword.id }}   foo > 10
+              foo
+            end
+            CRYSTAL
+        end
+      end
+    {% end %}
+
+    context "case" do
+      it "reports if there is redundant next in expressions" do
+        source = expect_issue subject, <<-CRYSTAL
+          case (foo = @foo)
+             # ^^^^^^^^^^^^ error: Redundant parentheses
+          when String then "string"
+          when Symbol then "symbol"
+          end
+          CRYSTAL
+
+        expect_correction source, <<-CRYSTAL
+          case foo = @foo
+          when String then "string"
+          when Symbol then "symbol"
+          end
+          CRYSTAL
+      end
+    end
+
+    it "reports correct rule, error message and position" do
+      s = Source.new %(
+        if (foo == 42)
+          foo
+        end
+      ), "source.cr"
+      subject.catch(s).should_not be_valid
+      s.issues.size.should eq 1
+      issue = s.issues.first
+      issue.rule.should_not be_nil
+      issue.location.to_s.should eq "source.cr:1:4"
+      issue.end_location.to_s.should eq "source.cr:1:14"
+      issue.message.should eq "Redundant parentheses"
+    end
```

Comment:
this kind of spec was needed just to test the error messages and issue locations. Since we introduced `expect_issue` spec helper this is not relevant anymore. We might need to remove these specs from all other places (which is out of scope of this PR), but for now my suggestion is to not add it to the newly added specs